Source/MQTTnet/Client/MqttClient.cs

```diff
@@ -38,8 +38,16 @@ public class MqttClient : Disposable, IMqttClient
 
         IMqttChannelAdapter _adapter;
         bool _cleanDisconnectInitiated;
-        long _isDisconnectPending;
-        bool _isConnected;
+        volatile int _connectState;
+
+        enum ConnectState
```

Comment:
Please move this to a dedicated c# file. In this project we do not have any private classes or enums etc. embedded in other classes. And consider renaming this to "MqttClientConnectionState" which follows the naming pattern in this project.

Reply:
I can do this tomorrow, but the name is really easy to be confused with `MQTTnet.Server.MqttClientConnectionStatus`

Reply:
You are right. Maybe we need to use the same name but due to a different namespace it is clear that the scope is a different one?

Reply:
Moved to `MQTTnet.Client.MqttClientConnectionStatus` and updated related field/method names.